test/src/org/labkey/test/tests/targetedms/TargetedMSMAMTest.java

```diff
@@ -0,0 +1,533 @@
+        // Verify expected peptides/proteins in the nested view
+        //Verify that amino acids from peptides are highlighted in blue as expected.
+        assertElementPresent(Locator.xpath("//tr//td//a//span[text()='LTSLNVVAGSDL'][span[contains(@style,'font-weight:bold;color:#0000ff;') and text()='R']]"));
+
+        if(smallMolPresent)
+        {
+           assertTextPresent("Small Molecule Precursor List");
+           assertTextPresent("Acylcarnitine");
+        }
+    }
+
+    @LogMethod
+    protected void verifyProteinSearch()
+    {
+        clickAndWait(Locator.linkContainingText("Panorama Dashboard"));
+        assertTextPresent("Mass Spec Search", "Protein Search");
+        _ext4Helper.clickExt4Tab("Protein Search");
+        waitForElement(Locator.name("identifier"));
+
+        // Test fix for issue 18217
+        // MRMer.zip contains the protein YAL038W.  MRMer_renamed_protein.zip contains the same protein with the
+        // name YAL038W_renamed.  MRMer.zip is imported first and  a new entry is created in prot.sequences with YAL038W
+        // as the bestname. MRMer_renamed_protein is imported second, and an entry is created in prot.identifiers
+        // for YAL038W_renamed. A search for YAL038W_renamed should return one protein result.
+        setFormElement(Locator.name("identifier"), "YAL038W_renamed");
+        waitAndClickAndWait(Locator.lkButton("Search"));
+        waitForText("Protein Search Results");
+        //waitForText("1 - 7 of 7");
+        assertTextPresentInThisOrder("Protein Search", "Matching Proteins (1)", "Targeted MS Proteins");
+
+        clickAndWait(Locator.id("expandCollapse-ProteinSearchProteinMatches"), 0); // Search results are hidden by default.
+        DataRegionTable potentialProteins = new DataRegionTable("PotentialProteins", this);
+        assertEquals(1, potentialProteins.getDataRowCount());
+        String bestName = potentialProteins.getDataAsText(0, "BestName");
+        assertTrue("Protein name didn't include 'YAL038W' :" + bestName, bestName.contains("YAL038W"));
+
+        DataRegionTable targetedMSMatches = new DataRegionTable("TargetedMSMatches", this);
+        List<String> labels = targetedMSMatches.getColumnDataAsText("Protein / Label");
+        assertEquals(0, labels.lastIndexOf("YAL038W"));
+        assertEquals(1, labels.indexOf("YAL038W_renamed"));
+    }
+
+    @LogMethod
+    protected void verifyModificationSearch()
+    {
+        // add modification search webpart and do an initial search by AminoAcid and DeltaMass
+        clickAndWait(Locator.linkContainingText("Panorama Dashboard"));
+        waitForElement(Locator.id("identifierInput"));
+        _ext4Helper.clickExt4Tab("Modification Search");
+        waitForElement(Locator.name("aminoAcids"));
```

Comment:
looks like strings "aminoAcids", "deltaMass", "unimodName", "customName"  are used in multiple places. can be pulled into variables?